src/genie/libs/parser/asa/show_inventory.py

```diff
@@ -49,14 +49,13 @@ def cli(self, output=None):
 
         # Name: "Chassis", DESCR: "ASA 5555-X with SW, 8 GE Data, 1 GE Mgmt"
         # Name: "power supply 1", DESCR: "ASA 5545-X/5555-X AC Power Supply"
-        p1 = re.compile(
-            r'^Name: +"+(?P<name>.+)"+,* +DESCR:+ "+(?P<description>.+)+"$')
+        p1 = re.compile(r'^Name:\s+"+(?P<name>.+)",\s+DESCR:\s"+(?P<description>.+)"$')
 
         # PID: ASA5555, VID: V01, SN: AAAAA11111
         # PID: AAA-AAA-AAA, VID: N/A, SN: AAA111
         # PID: N/A, VID: N/A, SN: AAAAA11111
-        p2 = re.compile(r'^PID: +(?P<pid>.+)( )?,+ VID: (?P<vid>.+)( )?, '
-            '+SN: (?P<sn>.+)$')
+        # PID: N/A, VID: , SN: AAAAA11111
+        p2 = re.compile(r'^PID:\s+(?P<pid>.+),+\sVID:(?P<vid>.+),\s+SN:(?P<sn>.+)$')
```

Comment:
This captures the space with the keys, which you remove in the for loop. It is better capture correctly the first time. I think it's fine to leave the space replacement in, it will matter for when only a space is captured, but I have altered the regex to not capture the extra spaces.

```suggestion
        p2 = re.compile(r'^PID:\s+(?P<pid>.+),\s+VID:\s?(?P<vid>.+),\s+SN:\s+(?P<sn>.+)$')
```

Reply:
I dont know what you mean by captures the spaces with the keys, the one I removed the one which was not capturing the empty output for VID: , I don't see much difference between your regex and mine, I tested on both now and had same results.